File: Python/Multiply Strings.py
class Solution:
    # @param num1, a string
    # @param num2, a string
    # @return a string
    def multiply(self, num1, num2):
        if num1 == "0" or num2 == "0":
            return "0"
        len1 = len(num1)
        len2 = len(num2)
        num = [0] * (len1 + len2)
        for i in range(len1):
            for j in range(len2):
                num[i+j+1] += int(num1[i]) * int(num2[j])
        for i in range(len1 + len2):
            j = len1 + len2 - i - 1
            if num[j] >= 10:
                num[j-1] += num[j] / 10
                num[j] = num[j] % 10
        res = 0
        while res < len1 + len2:
            if num[res] != 0:
                break
            res += 1
        st = ""
        for i in range(res,len1 + len2):
            st += str(num[i])
        return st

File: C++/Multiply-Strings.cpp
#include <string>

using namespace std;

string multiply(string num1, string num2) {
    string str;
    if(num1=="0"||num2=="0") return str="0";
    int len1=num1.size(),len2=num2.size();
    int *arry=new int[len1+len2];
    for(int i=0;i<len1+len2;i++)
        arry[i]=0;
    for(int i=0;i<len1;i++){
        for(int j=0;j<len2;j++){
            arry[i+j+1]=(num1[i]-'0')*(num2[j]-'0')+arry[i+j+1];
        }
    }
    for(int i=len1+len2-1;i>=0;i--){
        if(arry[i]>=10){
            arry[i-1]=arry[i-1]+arry[i]/10;
            arry[i]=arry[i]%10;
        }
    }
    int i;
    for(i=0;i<len1+len2;i++){
        if(arry[i]!=0) break;
    }
    for(i;i<len1+len2;i++){
        char ch='0'+arry[i];
        str=str+ch;
    }
    delete[] arry;
    return str;
}